sycl/include/sycl/interop_handle.hpp

```diff
@@ -15,13 +15,13 @@
 #include <sycl/detail/export.hpp>     // for __SYCL_EXPORT
 #include <sycl/detail/helpers.hpp>    // for context_impl
 #include <sycl/detail/impl_utils.hpp> // for getSyclObjImpl
-#include <sycl/detail/pi.h>           // for _pi_mem, pi_native_...
 #include <sycl/device.hpp>            // for device, device_impl
 #include <sycl/exception.hpp>
-#include <sycl/exception_list.hpp>    // for queue_impl
+#include <sycl/exception_list.hpp>                    // for queue_impl
 #include <sycl/ext/oneapi/accessor_property_list.hpp> // for accessor_property_list
 #include <sycl/image.hpp>                             // for image
 #include <sycl/properties/buffer_properties.hpp>      // for buffer
+#include <ur_api.h> // for ur_mem_handle_t, ur...
```

Comment:
I'm very concerned about includes like this. `ur_api.h` is almost 12k LOC long - we may not see it in our E2E tests, because they use some internal "reduced" SYCL header, but compilation of real-life applications will suffer from this. And here we are including it simply to get a couple of typedefs.

We need some in-tree header which would provide necessary interfaces to us without including the whole UR. Moreover, I think that we need to develop a test to prohibit direct inclusion of `ur_` headers outside of `source/` folder to prevent any potential compile-time regressions.

Reply:
I understand the concern here. I think doing this would make most sense being part of UR. We would need some time to sort out the header generation scripts to do this correctly, though. I've created https://github.com/oneapi-src/unified-runtime/issues/1890 to track the UR side of this.